eng/scripts/CodeCheck.ps1

```diff
@@ -190,43 +190,6 @@ try {
             & git --no-pager diff --ignore-space-change $filePath
         }
     }
-
-    $targetBranch = $env:SYSTEM_PULLREQUEST_TARGETBRANCH
-
-    if (![string]::IsNullOrEmpty($targetBranch)) {
-        if ($targetBranch.StartsWith('refs/heads/')) {
-            $targetBranch = $targetBranch.Replace('refs/heads/','')
-        }
-
-        # Retrieve the set of changed files compared to main
-        Write-Host "Checking for changes to API baseline files $targetBranch"
-
-        $changedFilesFromTarget = git --no-pager diff origin/$targetBranch --ignore-space-change --name-only --diff-filter=ar
-        $changedAPIBaselines = [System.Collections.Generic.List[string]]::new()
-
-        if ($changedFilesFromTarget) {
-            foreach ($file in $changedFilesFromTarget) {
-                # Check for changes in Shipped in all branches
-                if ($file -like '*PublicAPI.Shipped.txt') {
-                    $changedAPIBaselines.Add($file)
-                }
-                # Check for changes in Unshipped in servicing branches
-                if ($targetBranch -like 'release*' -and $file -like '*PublicAPI.Unshipped.txt') {
-                    $changedAPIBaselines.Add($file)
-                }
-            }
-        }
-
-        Write-Host "Found changes in $($changedAPIBaselines.count) API baseline files"
-
-        if ($changedAPIBaselines.count -gt 0) {
-            LogError "Detected modification to baseline API files. PublicAPI.Shipped.txt files should only be updated after a major release. See /docs/APIBaselines.md for more information."
```

Comment:
According to @sharwell  / @mavasani, updating the PublicAPI.txt is the expected behavior when nullability annotations have changed. See https://github.com/dotnet/roslyn-analyzers/issues/3253. 

By the way, I tried following the instructions https://github.com/dotnet/aspnetcore/blob/main/docs/APIBaselines.md#updated-apis here but it just does not work with Mvc.Core (you're free to try it with this PR). Letting the tool decide what's best for itself appears to be the best course of action.

Reply:
No

Reply:
In a few more words: We want the PublicApi.Unshipped.txt files to reflect all changes since 5.0. Anything other than adding empty PublicAPI.Shipped.txt files is not something we're going to do in the main branch.

Reply:
Actually I'm missing something. You're not changing any PublicApi.Shipped.txt files. This check shouldn't break anything as this branch is now.

Reply:
BTW I disagree strongly with a quote from dotnet/roslyn-analyzers#3253.
> The annotated API has not shipped, but the API contract has shipped and it doesn't change by the addition of nullable annotations

If that were the case, we would not have a breaking change announcement tracking our nullability additions and changes.

Reply:
I've set aside some time to talk to @JunTaoLuo about this today. I wasn't able to get `*removed*` to work in some cases maybe he's able to help.

Reply:
@JunTaoLuo was able to help me figure this out.

Reply:
Great to hear❕ @pranavkm and @JunTaoLuo are https://github.com/dotnet/aspnetcore/blob/main/docs/APIBaselines.md needed to avoid this recurring❔